### PR TITLE
Fix a test flake

### PR DESCRIPTION
**Pull request by adrianludwin**
Observed this flake on #703, made it retryable.

Tested: ran the tests six times, they all passed

### CODE REVIEW

File: incubator/hnc/pkg/reconcilers/anchor_test.go [hunk 1]
```diff
@@ -75,10 +75,13 @@ var _ = Describe("Anchor", func() {
...
 		// update of the HC instance (hc.Spec.AllowCascadingDelete).
-		barHier := getHierarchy(ctx, barName)
-		barHier.Spec.Parent = "other"
-		barHier.Spec.AllowCascadingDelete = true
-		updateHierarchy(ctx, barHier)
+		Eventually(func() bool {
+			barHier := getHierarchy(ctx, barName)
+			barHier.Spec.Parent = "other"
+			barHier.Spec.AllowCascadingDelete = true
+			updateHierarchy(ctx, barHier) // can fail if made too quickly
```
Comment (yiqigao217) [inline incubator/hnc/pkg/reconcilers/anchor_test.go:82]: Hmm, seems the test is still flaky. Was the error you got is something like "the object is updated, please update on a new version"?

/hold

**Comment by adrianludwin** [comment]
Yup. I'll have another look.

**Comment by adrianludwin** [comment]
Looks like you can't nest "Eventually"s. Fixed in a better way.

**Comment by yiqigao217** [comment]
> Looks like you can't nest "Eventually"s. Fixed in a better way.

Did you run the test more than once? If so maybe update the commit/PR message?

**Comment by adrianludwin** [comment]
> Did you run the test more than once? If so maybe update the commit/PR message?

No, only once. It's so slow :(

**Comment by adrianludwin** [comment]
Sigh, you're right :D Set it up to run five times, but heading out for dinner now. You can wait to approve it you want!

**Comment by adrianludwin** [comment]
Updated comment, thanks for making me do the right thing!

**Pull request merged**